### PR TITLE
Update Rector to version 0.18

### PR DESCRIPTION
**Pull request by leofeyer**

### CODE REVIEW

File: core-bundle/src/EventListener/CommandSchedulerListener.php [hunk 1]
```diff
@@ -55,11 +55,7 @@ private function shouldRunCron(Request $request): bool
             return false;
         }
 
-        if ($this->autoMode && $this->cron->hasMinutelyCliCron()) {
-            return false;
-        }
-
-        return true;
+        return !($this->autoMode && $this->cron->hasMinutelyCliCron());
```
Comment (leofeyer) [inline core-bundle/src/EventListener/CommandSchedulerListener.php]: @ausi Would `return !$this->autoMode || !$this->cron->hasMinutelyCliCron();` be the same?

File: core-bundle/src/HttpKernel/ModelArgumentResolver.php [hunk 2]
```diff
@@ -44,11 +44,7 @@ public function supports(Request $request, ArgumentMetadata $argument): bool
             return false;
         }
 
-        if (!$argument->isNullable() && null === $this->fetchModel($request, $argument)) {
-            return false;
-        }
-
-        return true;
+        return !(!$argument->isNullable() && null === $this->fetchModel($request, $argument));
```
Comment (leofeyer) [inline core-bundle/src/HttpKernel/ModelArgumentResolver.php]: @ausi Would `return $argument->isNullable() || null !== $this->fetchModel($request, $argument);` be the same?

File: core-bundle/src/EventListener/CommandSchedulerListener.php (hunk 1 above)
Comment (ausi) [reply core-bundle/src/EventListener/CommandSchedulerListener.php]: According to [De Morgan’s laws](https://en.wikipedia.org/wiki/De_Morgan%27s_laws), yes 🤓

File: core-bundle/src/HttpKernel/ModelArgumentResolver.php (hunk 2 above)
Comment (ausi) [reply core-bundle/src/HttpKernel/ModelArgumentResolver.php]: Yes.

File: core-bundle/src/EventListener/CommandSchedulerListener.php (hunk 1 above)
Comment (ausi) [reply core-bundle/src/EventListener/CommandSchedulerListener.php]: But the old version with `if ($this->autoMode && $this->cron->hasMinutelyCliCron()) {` is way more readable IMO.

Comment (leofeyer) [reply core-bundle/src/EventListener/CommandSchedulerListener.php]: Reverted in f8fc72b51bd70224eb51a04d462fc2e3ce04206b.

File: core-bundle/src/HttpKernel/ModelArgumentResolver.php (hunk 2 above)
Comment (leofeyer) [reply core-bundle/src/HttpKernel/ModelArgumentResolver.php]: Also reverted in f8fc72b51bd70224eb51a04d462fc2e3ce04206b.

**Pull request merged**